internal/blockchain/ethereum/ethsigner/ethsigner.go

```diff
@@ -0,0 +1,209 @@
+// Copyright © 2022 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ethsigner
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/hyperledger/firefly-cli/internal/blockchain/ethereum"
+	"github.com/hyperledger/firefly-cli/internal/constants"
+	"github.com/hyperledger/firefly-cli/internal/docker"
+	"github.com/hyperledger/firefly-cli/internal/log"
+	"github.com/hyperledger/firefly-cli/pkg/types"
+)
+
+var ethsignerImage = "ghcr.io/hyperledger/firefly-signer:head"
+var gethImage = "ethereum/client-go:release-1.10"
+
+const useJavaSigner = false // also need to change the image appropriately if you recompile to use the Java signer
+
+type EthSignerProvider struct {
```

Comment:
Is FireFly Signer also an Ethsigner? Or should we rename these things `signerImage` and `SignerProvider` to convey the fact that they could be Ethsigner or FF Signer?

Reply:
I can rename the whole package to `signer` for sure, to get rid of the `eth` prefix

Reply:
However, I had migration concerns - and didn't think the `eth` prefix was inaccurate... which is why I left it.

Reply:
I understand the Git repo that provides the Ethereum signer runtime, is called `firefly-signer`... which was deliberate because I think that repo will be used for non-ethereum based EVM chains in the future.

Reply:
Sorry for the ramble - net is I think this *is* an `EthSignerProvider`, and I think `ethsignerImage` is the right name.

Reply:
I have realized I meant to move this from `head` to the `v1.0.1` image - pushing that change

Reply:
Okay I'm fine with this. Thanks for the extra details 🙂